pkg/nspawntool/run.go

```diff
@@ -83,6 +84,23 @@ func RunNode(k8srelease, name string) error {
 	// defaultBinds has to be determined after evaluation of cniPath
 	defaultBinds = getDefaultBinds(cniPath)
 
+	if kubeSpawnDir == "" {
+		kubeSpawnDir = parseBind("$PWD/.kube-spawn")
+	} else if err := utils.CheckValidDir(kubeSpawnDir); err != nil {
+		kubeSpawnDir = parseBind("$PWD/.kube-spawn")
+	}
+
+	if err := bootstrap.PathSupportsOverlay(kubeSpawnDir); err != nil {
+		return fmt.Errorf("try to pass a directory with a different filesystem (like ext4 or XFS) to --kube-spawn-dir.")
```

Comment:
Sorry, I ninja-edited the comment before and I guess you read the old version. I'd rather get a bit more context by getting the error from `PathSupportsOverlay()`:

```
return fmt.Errorf("unable to create overlayfs on %q: %v. Try to pass a directory with a different filesystem (like ext4 or XFS) to --kube-spawn-dir.
```